integration_tests/src/main/python/arithmetic_ops_test.py

```diff
@@ -498,3 +500,40 @@ def test_greatest(data_gen):
             lambda spark : gen_df(spark, gen).select(
                 f.greatest(*command_args)), conf=allow_negative_scale_of_decimal_conf)
 
+
+def _spark_33008_datagen(spark_session: SparkSession) -> DataFrame:
+    return spark_session.createDataFrame([
+        (1, 2),
+        (1, 0)    # arithmetic exception for _1 / _2
+    ])
+
+
+def _assert_div_by_zero(session_func, test_div_func, conf):
+    with pytest.raises(Py4JJavaError) as py4jError:
+        session_func(test_div_func, conf)
+    assert 'java.lang.ArithmeticException: divide by zero' in str(py4jError.value.java_exception)
+
+@pytest.mark.parametrize('from_disk', [False, True])
+@pytest.mark.parametrize('ansi_mode', [False, True])
+def test_spark_33008(spark_tmp_path, from_disk, ansi_mode):
+    if ansi_mode and is_before_spark_310():
+        pytest.xfail('https://github.com/apache/spark/pull/29882')
+
+    ansi_conf = {'spark.sql.ansi.enabled': ansi_mode}
+
+    if from_disk:
```

Comment:
Why is there an option to round-trip through Parquet for this test?  I'm not sure how that's supposed to impact divide-by-zero testing.  If it's necessary, why not ORC, CSV, etc?  If it's not necessary then let's remove it and save on testing time.

Reply:
It was a more of a speculation on my part that it may play a role, so I wanted to make sure that both cases are covered. 

So far through the test I learned that there is no difference w.r.t to that. It helps me learn the code pieces exercised in both cases. I'll remove it after proving it does not add any value